File: 이건희/level2/타겟 넘버.py
def solution(numbers, target):
    result = 0
    
    def dfs(sumNum, lenNum):
        nonlocal result
        
        if lenNum == len(numbers): 
            if sumNum == target:
                result += 1
            return
        
        num = [-sumNum, sumNum]
        
        if lenNum == 1:
            for i in range(2):
                dfs(num[i] + numbers[lenNum], lenNum + 1)
                dfs(num[i] - numbers[lenNum], lenNum + 1)
        else:
            dfs(sumNum + numbers[lenNum], lenNum + 1)
            dfs(sumNum - numbers[lenNum], lenNum + 1)
            
    dfs(numbers[0], 1)
    return result

File: 김찬우/프로그래머스 /Lv.2/타겟넘버.java
class Solution {
    
    static int count;
    
    public int solution(int[] numbers, int target) {
        
        count = 0;
        int answer;
        dfs(numbers,target,0,0);
        answer = count;
        
        return answer;
    }
    
    static void dfs(int[] numbers,int target, int depth, int sum){
        
        if(numbers.length == depth){
            if(sum == target){
                count++;
                
            }
            return;
        }
        
        int plus = numbers[depth] + sum;
        int minus = sum - numbers[depth];
        
        dfs(numbers,target,depth+1,plus);
        dfs(numbers,target,depth+1,minus);
        
    }
}
